src/lib/falcor/falcorUtils.js

```diff
@@ -124,6 +128,28 @@ export function pathSetsInCache(cache, falcorPathSets) {
     return nextKeySet.every((keyOrRange) => {
       if (keyOrRange !== null && typeof keyOrRange === "object") {
         // keyOrRange is a range
+
+        // Every time we call a range, there should also be a length property.
+        // This is so we know when overfetching whether data is missing
+        // in the cache or it's simply because there is no data to fetch.
+        // It is also needed in the software development to know how many
+        // items you will actually receive when overfetching.
+        if (!curObject.hasOwnProperty("length")) {
+          console.warn("No length property on object in cache. This might be a developer mistake.\
+ If error checking flag is true current object will be printed below.");
+          // As this will be printing a lot we don't want to always print full objects
+          // change the flag to true if you want to debug
+          if (false) {
```

Comment:
This should be the error checking flag, though I think we should just use the PRODUCTION flag instead for this.